src/ILCompiler.WebAssembly/src/Compiler/DependencyAnalysis/WebAssemblyVTableSlotNode.cs

```diff
@@ -45,7 +44,7 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
         {
             DependencyList result = new DependencyList();
 
-            if (!factory.VTable(_targetMethod.OwningType).HasFixedSlots)
+            if (!factory.VTable(_targetMethod.OwningType).HasFixedSlots && !_targetMethod.IsRuntimeDeterminedExactMethod)
```

Comment:
We shouldn't see runtime determined methods here (make it an assert in the constructor). The caller should grab the canonical method body before they ask for the WebAssemblyVTableSlotNode.